src/main/java/sirius/biz/mongo/MultiLanguageStringPropertyTransformer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.mongo;
+
+import sirius.db.mixing.properties.MultiLanguageStringProperty;
+import sirius.db.mixing.types.StringList;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.di.transformers.Transformer;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+
+/**
+ * Generates a {@link StringList} from the values of a {@link MultiLanguageStringProperty}.
+ */
+@Register
+public class MultiLanguageStringPropertyTransformer
+        implements Transformer<MultiLanguageStringProperty, PrefixSearchableContentSupplier> {
+
+    @Override
+    public Class<MultiLanguageStringProperty> getSourceClass() {
+        return MultiLanguageStringProperty.class;
+    }
+
+    @Override
+    public Class<PrefixSearchableContentSupplier> getTargetClass() {
+        return PrefixSearchableContentSupplier.class;
+    }
+
+    @Nullable
+    @Override
+    @SuppressWarnings("unchecked")
+    public PrefixSearchableContentSupplier make(@Nonnull MultiLanguageStringProperty source) {
+        return entity -> {
+            StringList tokens = new StringList();
```

Comment:
why this intermediate step, couldn't we transform into a BiConsumer which accepts an entity and a StringBuilder or even BiConsumer<PrefixSearchableEntity, Consumer<String>> and then invoke with (entity, this::addContentString) or the like

Reply:
or simply transform to String

Reply:
We index not only the words, but the phrase as well (see test case)
Indigo blue becomes: ["indigo blue", "indigo", "blue"]

For Multi-Language scenario we want each translation "phrase" to be processed independently and not only its loose words...

Reply:
Implemented as `BiConsumer`. Hopefully nailed it now 😬